### DISCUSSION TITLE
Coupled Variable Name

### DISCUSSION THREAD

**Discussion by rtaylo45** [idaholab/moose#18084] [Q&A General] [answered]
Im writing a kernel and that works with two coupled variables. I was wondering if there is a way to retrieve the name of a specific couple variable while inside the code for the kernel. For example, if my input file has this example kernel
 
```
[kernels]
  [exampleKernel]
    type = ExampleKernel
    variable = nonLinearVar
    coupled_var_1 = example_coupled_var_1
    coupled_var_2 = example_coupled_var_2
  []
[]
```

In my kernel i can use `_var.name()` to retrieve the name of the primal variable. But how would get the name of the associated inputs for `coupled_var_1` and `coupled_var_2`? Something like `_coupled_var_1.name()` and `_coupled_var_1.name()`. From examples i have seen with multi-physics coupling, the type of the coupled variable is `VariableValue` which is not a type which I can extract a variable name from.

**Comment by GiudGiud** [comment, marked as answer]
Hello

I would do getFieldVar("parameter_name", 0).name()  (might be ->name()) 

0 is the component, I'm assuming you are not using array variables. 
"parameter_name" is "coupled_var_i" here

Guillaume

**Reply by dschwen** [reply]
See here for example:

https://github.com/idaholab/moose/blob/06bc1f73f0d4dfb3d01ce9ba77d6bb97346e07eb/modules/tensor_mechanics/src/materials/ComputeVariableEigenstrain.C#L36

**Reply by rtaylo45** [reply]
Awesome, this works. Thanks for the help
